File: frontend/types/conversation.ts
export interface Conversation {
    id: number;

    application_id: number;

    created_at: string;
    updated_at: string;
}


export interface ConversationListItem {
    id: number;

    application_id: number;

    other_user_id: number;
    other_user_email: string;

    vacancy_id: number;
    vacancy_title: string;

    unread_count: number;

    last_message: string | null;
    last_message_at: string | null;

    created_at: string;
    updated_at: string;
}

File: backend/app/schemas/conversation.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ConversationCreate(BaseModel):
    application_id: int


class ConversationResponse(BaseModel):
    id: int

    application_id: int

    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )


class ConversationListItem(BaseModel):
    id: int

    application_id: int

    other_user_id: int
    other_user_email: str

    vacancy_id: int
    vacancy_title: str

    unread_count: int

    last_message: str | None
    last_message_at: datetime | None

    created_at: datetime
    updated_at: datetime